src/util/util.js

```diff
@@ -125,22 +125,34 @@ function extractURLsFromTemplates(URLTemplates) {
 }
 
 /**
- * Filter URLTemplates elements to keep only valid and safe URL templates.
+ * Filter URLTemplates elements .
  *   To be valid, urls should:
  *   - have the same protocol as the client
  *   or
  *   - be protocol-relative urls
  *
+ * Returns an object with two arrays
+ *    - validUrls : An array of valid URLs
+ *    - invalidUrls: An array of invalid URLs
+ *
  * @param {Array} URLTemplates - A Array of string/object containing urls templates.
+ * @returns {Object}
+ *
  */
-function filterValidUrlTemplates(URLTemplates) {
-  if (Array.isArray(URLTemplates)) {
-    return URLTemplates.filter(urlTemplate => {
-      const url = urlTemplate.hasOwnProperty('url') ? urlTemplate.url : urlTemplate;
-      return isValidUrl(url);
-    })
-  }
-  return isValidUrl(URLTemplates);
+function filterUrlTemplates(URLTemplates) {
+  return URLTemplates.reduce(
+    (acc, urlTemplate) => {
+      const url = urlTemplate.hasOwnProperty('url')
+        ? urlTemplate.url
+        : urlTemplate;
```

Comment:
In which case can we have an object instead ? 
```suggestion
      const url = urlTemplate.url || urlTemplate;
```